google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddRealEstateFeed.java

```diff
@@ -0,0 +1,507 @@
+            .setPlaceholderType(PlaceholderType.DYNAMIC_REAL_ESTATE)
+            .setFeed(StringValue.of(feedResourceName))
+            .addAttributeFieldMappings(listingIdMapping)
+            .addAttributeFieldMappings(listingNameMapping)
+            .addAttributeFieldMappings(finalUrlsMapping)
+            .addAttributeFieldMappings(imageUrlMapping)
+            .addAttributeFieldMappings(contextualKeywordsMapping)
+            .build();
+
+    // Creates the operation.
+    FeedMappingOperation operation =
+        FeedMappingOperation.newBuilder().setCreate(feedMapping).build();
+
+    // Adds the FeedMapping.
+    try (FeedMappingServiceClient feedMappingServiceClient =
+        googleAdsClient.getLatestVersion().createFeedMappingServiceClient()) {
+      MutateFeedMappingsResponse response =
+          feedMappingServiceClient.mutateFeedMappings(
+              Long.toString(customerId), ImmutableList.of(operation));
+
+      // Displays the results.
+      for (MutateFeedMappingResult result : response.getResultsList()) {
+        System.out.printf(
+            "Created feed mapping with resource name '%s'.%n", result.getResourceName());
+      }
+    }
+  }
+
+  /**
+   * Adds the new items to the feed.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @param feedAttributes the feed attributes.
+   * @param feedResourceName the feed resource name.
+   */
+  private void createFeedItems(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      Map<RealEstatePlaceholderField, FeedAttribute> feedAttributes,
+      String feedResourceName) {
+
+    // Creates the listing ID feed attribute value.
+    FeedItemAttributeValue listingId =
+        FeedItemAttributeValue.newBuilder()
+            .setFeedAttributeId(
+                Int64Value.of(
+                    feedAttributes.get(RealEstatePlaceholderField.LISTING_ID).getId().getValue()))
+            .setStringValue(StringValue.of("ABC123DEF"))
+            .build();
+    // Creates the listing name feed attribute value.
+    FeedItemAttributeValue listingName =
+        FeedItemAttributeValue.newBuilder()
+            .setFeedAttributeId(
+                Int64Value.of(
+                    feedAttributes.get(RealEstatePlaceholderField.LISTING_NAME).getId().getValue()))
+            .setStringValue(StringValue.of("Two bedroom with magnificent views"))
+            .build();
+    // Creates the final URLs feed attribute value.
+    FeedItemAttributeValue finalUrls =
+        FeedItemAttributeValue.newBuilder()
+            .setFeedAttributeId(
+                Int64Value.of(
+                    feedAttributes.get(RealEstatePlaceholderField.FINAL_URLS).getId().getValue()))
+            .addStringValues(StringValue.of("http://www.example.com/listings/"))
+            .build();
+
+    // Optionally insert additional attributes here, such as address, city, description, etc.
```

Comment:
Include a link to the docs.

Also note that if you set other fields here they'll need to be added to the mapping above.